### DISCUSSION TITLE
Actions in Form Fields

### DISCUSSION THREAD

**Discussion by Z3d0X** [filamentphp/filament#1897] [Help] [answered]
Is there anyway to include Actions in the form? Im attempting to do something like this
![image](https://user-images.githubusercontent.com/75579178/158106281-e7b8b916-8552-47ce-a94e-7c526f76f39e.png)

**Comment by studiowizjo** [comment]
In **CreateRecord** or **EditRecord** resource you can use method:

```php
protected function getFormActions(): array
{
  return [
    $this->myCustomAction(),
    $this->getCreateFormAction(),
    $this->getCreateAndCreateAnotherFormAction(),
    $this->getCancelFormAction(),
  ];
}

protected function myCustomAction(): ButtonAction
{
  return ButtonAction::make('save')
    ->label('Label')
    ->action(function() {
      //your logic
    });
}	
```

but this will render button at the bottom of form. If you need it to be in the middle, I think the only way would be [creating your own field](https://filamentphp.com/docs/2.x/forms/fields#building-custom-fields)

**Comment by danharrin** [comment, marked as answer]
#2057
